### ISSUE TITLE
[WAV]In some cases, information can't be acquired with a file other than the PCM format.

### ISSUE THREAD

**Issue by rai1029ro** [taglib/taglib#850]
Hello.

WAV Spec is PCM and Non-PCM and Extensible.
（info:http://www-mmsp.ece.mcgill.ca/Documents/AudioFormats/WAVE/WAVE.html)

1. The non-PCM formats must have a fact chunk
However,
2. ExntesibleFormats does't seem to be required a fact chunk.

taglib can't get information, because fact chunk is mandatory except for PCM format.

（ex)
http://www-mmsp.ece.mcgill.ca/Documents/AudioFormats/WAVE/Samples.html
is "WAVE (WAVE_FORMAT_EXTENSIBLE) file sample"
or 
http://www.e-onkyo.com/music/ Hi-res File）

Best regards

**Comment by xiuyusong** [comment]
I met the same problem, it seems that the line 185 in wavproperties.cpp should add more check
  d->format = data.toShort(0, false);
if (d->format == 0xfffe) {
  // if extensible then read the format from the subformat
  d->format = data.toShort(24, false);
}
  if(d->format != FORMAT_PCM && totalSamples == 0) {
    debug("RIFF::WAV::Properties::read() - Non-PCM format, but 'fact' chunk not found.");
    return;
  }

**Comment by scjhan** [comment]
I solved the same problem with following code:
```c++
if(d->format != FORMAT_PCM && totalSamples == 0) {
    debug("RIFF::WAV::Properties::read() - Non-PCM format, but 'fact' chunk not found.");
    if (data.toShort(16, false) == 0) {
        debug("RIFF::WAV::Properties::read() - Extra format bytes is 0, read as PCM format");
        d->format = FORMAT_PCM;
    }
    else {
        return;
    }
}
```

**Comment by ufleisch** [comment]
Fixed with #986.
